packages/slo-stat-bot/src/slo-stat-label.ts

```diff
@@ -0,0 +1,221 @@
+      const pull_number = context.payload.number;
+
+      const fileResponse:
+        | Response<PullsListFilesResponse>
+        | undefined = await handler.listFiles(
+        context.github,
+        owner,
+        repo,
+        pull_number,
+        100
+      );
+      let fileList: PullsListFilesResponseItem[] | undefined;
+
+      if (fileResponse !== undefined) fileList = fileResponse.data;
+
+      for (
+        let i = 0;
+        fileList !== undefined && fileList[i] !== undefined;
+        i++
+      ) {
+        const file = fileList[i];
+        if (
+          file.filename === '.github/issue_slo_rules.json' ||
+          (repo === '.github' && file.filename === 'issue_slo_rules.json')
+        ) {
+          await handler.handle_slos(
+            context,
+            owner,
+            repo,
+            pull_number,
+            file.sha
+          );
+          break;
+        }
+      }
+    }
+  );
+}
+
+handler.handle_slos = async function handle_slos(
+  context: Context,
+  owner: string,
+  repo: string,
+  issue_number: number,
+  file_sha: string
+) {
+  const sloString = await handler.getFileContents(
+    context.github,
+    owner,
+    repo,
+    file_sha
+  );
+
+  if (sloString !== undefined) {
+    const sloData = JSON.parse(sloString);
+    const res: ValidationResults = await handler.lint(schema, sloData);
+
+    await handler.commentPR(
+      context.github,
+      owner,
+      repo,
+      issue_number,
+      res.isValid
+    );
+    await handler.createCheck(context, res);
+  }
+};
+
+handler.getFileContents = async function getFileContents(
+  github: GitHubAPI,
+  owner: string,
+  repo: string,
+  file_sha: string
+): Promise<string | undefined> {
+  try {
+    const blob = await github.git.getBlob({
+      owner,
+      repo,
+      file_sha,
+    });
+    const fileContent = Buffer.from(blob.data.content, 'base64').toString(
+      'utf8'
+    );
+    return fileContent;
+  } catch (err) {
+    console.log(err);
+    return;
+  }
+};
+
+handler.listFiles = async function listFiles(
+  github: GitHubAPI,
+  owner: string,
+  repo: string,
+  pull_number: number,
+  per_page: number
+): Promise<Response<PullsListFilesResponse> | undefined> {
+  try {
+    const listOfFiles = await github.pulls.listFiles({
+      owner,
+      repo,
+      pull_number,
+      per_page,
+    });
+    return listOfFiles;
+  } catch (err) {
+    console.log(err);
+    return;
+  }
+};
+
+handler.lint = async function lint(
+  schema: JSON,
+  sloData: JSON
+): Promise<ValidationResults> {
+  const ajv = new Ajv();
+  const validate = await ajv.compile(schema);
+  const isValid = await validate(sloData);
+
+  return {
+    isValid: isValid,
+    errors: validate.errors,
+  } as ValidationResults;
+};
+
+handler.commentPR = async function commentPR(
+  github: GitHubAPI,
+  owner: string,
+  repo: string,
+  issue_number: number,
+  isValid: boolean
+) {
+  if (!isValid) {
```

Comment:
This is a place where I'd flip the condition of the `if` to keep the code more flat.  You could say something like:

```js
if (isValid) {
  return;
}
```

Then you could go about the rest of the function without being inside of an `if` clause.